### PR TITLE
Creating 'global constant' to fix the issue with BigInt

### PR DESCRIPTION
**Pull request by DmitryVasilevsky**
This code change allows large BigInts to be defined literally.
Previously only BigInts small enough to fit into int64 were supported. A code path that was supposed to support larger BigInts was calling GetElementPtr on a non-pointer type (an array type). It was causing a failure. I updated the code to be similar to handling string literals. Now a global internal variable is created to represent a large BigInt. This variable has pointer type so the code succeeds.

NOTE: C++ QIR runtime still doesn't support BigInts.

Fixes https://github.com/microsoft/qsharp-compiler/issues/1046

### CODE REVIEW

File: src/QsCompiler/QirGeneration/Context.cs [hunk 1]
```diff
@@ -481,6 +481,17 @@ internal GlobalVariable GetOrCreateStringConstant(string str)
             return constant;
         }
 
+        /// <summary>
+        /// Creates a global constant (data array) that stores the given bytes.
+        /// </summary>
+        internal GlobalVariable CreateArrayConstant(byte[] array) {
```
Comment (bettinaheim) [inline src/QsCompiler/QirGeneration/Context.cs:487]: How about naming it CreateGlobalConstantArray, to be a bit more descriptive?

Comment (DmitryVasilevsky) [reply src/QsCompiler/QirGeneration/Context.cs]: Yes, this is a much better name! (I didn't like the previous name, just named it similar to the string function.)

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBigInts.qs [hunk 2]
```diff
@@ -16,5 +16,8 @@ namespace Microsoft.Quantum.Testing.QIR
     @EntryPoint()
     function Main() : Unit {
         let _ = TestBigInts(0L, 0L);
+        let N1 = 12345L;
```
Comment (bettinaheim) [inline src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBigInts.qs:19]: It looks like the code generation only checks the generated IR for TestBigInts. Could you hence either add these bigint instantiations to the TestBigInts function instead, or add a file to also check the generated .ll code for the Main function?

Comment (DmitryVasilevsky) [reply src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestBigInts.qs:19]: Added another file to test Main. Changed qirTest to qirMultiTest.

**Review by bettinaheim — APPROVED** [review]
It would be good to confirm that it runs, but other than that lgtm.

**Pull request merged**